group-coordinator/src/main/java/org/apache/kafka/coordinator/group/OffsetConfig.java

```diff
@@ -20,16 +20,53 @@
+            "also when a topic is deleted via the delete-topic request, upon propagated metadata update any group's committed offsets for that topic will also be deleted without extra retention period.";
+
+    public static final String OFFSETS_RETENTION_CHECK_INTERVAL_MS_CONFIG = "offsets.retention.check.interval.ms";
+    public static final long OFFSETS_RETENTION_CHECK_INTERVAL_MS_DEFAULT = 600000L;
+    public static final String OFFSETS_RETENTION_CHECK_INTERVAL_MS_DOC = "Frequency at which to check for stale offsets";
+
+    public static final String OFFSETS_TOPIC_PARTITIONS_CONFIG = "offsets.topic.num.partitions";
+    public static final int OFFSETS_TOPIC_PARTITIONS_DEFAULT = 50;
+    public static final String OFFSETS_TOPIC_PARTITIONS_DOC = "The number of partitions for the offset commit topic (should not change after deployment).";
+
+    public static final String OFFSETS_TOPIC_SEGMENT_BYTES_CONFIG = "offsets.topic.segment.bytes";
+    public static final int OFFSETS_TOPIC_SEGMENT_BYTES_DEFAULT = 100 * 1024 * 1024;
+    public static final String OFFSETS_TOPIC_SEGMENT_BYTES_DOC = "The offsets topic segment bytes should be kept relatively small in order to facilitate faster log compaction and cache loads.";
+
+    public static final String OFFSETS_TOPIC_REPLICATION_FACTOR_CONFIG = "offsets.topic.replication.factor";
+    public static final short OFFSETS_TOPIC_REPLICATION_FACTOR_DEFAULT = 3;
+    public static final String OFFSETS_TOPIC_REPLICATION_FACTOR_DOC = "The replication factor for the offsets topic (set higher to ensure availability). " +
+            "Internal topic creation will fail until the cluster size meets this replication factor requirement.";
+
+    public static final String OFFSETS_TOPIC_COMPRESSION_CODEC_CONFIG = "offsets.topic.compression.codec";
+    public static final CompressionType OFFSETS_TOPIC_COMPRESSION_CODEC_DEFAULT = CompressionType.NONE;
+    public static final String OFFSETS_TOPIC_COMPRESSION_CODEC_DOC = "Compression codec for the offsets topic - compression may be used to achieve \"atomic\" commits.";
+
+    public static final String OFFSET_COMMIT_TIMEOUT_MS_CONFIG = "offsets.commit.timeout.ms";
+    public static final int OFFSET_COMMIT_TIMEOUT_MS_DEFAULT = 5000;
+    public static final String OFFSET_COMMIT_TIMEOUT_MS_DOC = "Offset commit will be delayed until all replicas for the offsets topic receive the commit " +
+            "or this timeout is reached. This is similar to the producer request timeout.";
+
+    public static final String OFFSET_COMMIT_REQUIRED_ACKS_CONFIG = "offsets.commit.required.acks";
+    public static final short OFFSET_COMMIT_REQUIRED_ACKS_DEFAULT = -1;
+    public static final String OFFSET_COMMIT_REQUIRED_ACKS_DOC = "The required acks before the commit can be accepted. In general, the default (-1) should not be overridden.";
+
+    public static final long OFFSET_RETENTION_MS_DEFAULT = 24 * 60 * 60 * 1000L;
```

Comment:
this default value is not equal to `OFFSETS_RETENTION_MINUTES_DEFAULT`. that is weird to me ...

Reply:
yeah this is weird also  because `OFFSET_RETENTION_MS_DEFAULT` has only be used as a default value for `OffsetConfig` constructor in the original scala code https://github.com/apache/kafka/pull/15161/files#diff-157be940464ab3026e373cbc03b515adc52cdc315aba96ecf384e1db85a93f29. While `GroupCoordinator::offsetConfig` was setting this value to `config.offsetsRetentionMinutes * 60L * 1000L`.

I'll delete it as seems like no one is actually referring to this default value